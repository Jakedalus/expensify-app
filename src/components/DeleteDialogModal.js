import React from 'react';
import Modal from 'react-modal';

const DeleteDialogModal = (props) => (
    <Modal
        isOpen={props.clickedDelete}
        onRequestClose={props.handleClearDeleteDialogModal}
        closeTimeoutMS={200}
        contentLabel="Modal"
        className="modal"
    >
        <h2>Do you want to delete this expense: {props.description}?</h2>
        <button className="button button--secondary" onClick={() => props.handleClearDeleteDialogModal(true)}>Delete</button>
        <button className="button button--secondary" onClick={() => props.handleClearDeleteDialogModal(false)}>Cancel</button>
    </Modal>
);

export default DeleteDialogModal;