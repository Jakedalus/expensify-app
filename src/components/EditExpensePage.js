import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import DeleteDialogModal from './DeleteDialogModal';
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      clickedDelete: false
    };
  }

  handleClearDeleteDialogModal = (confirmDelete) => {
    if(confirmDelete) {
      this.props.startRemoveExpense({ id: this.props.expense.id });
      this.props.history.push("/");
    } 
    
    this.setState(() => ({clickedDelete: false})); 
  };

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  onClick = () => {
    this.setState(() => ({ clickedDelete: true }));
    // this.props.startRemoveExpense({ id: this.props.expense.id });
    // this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="button button--secondary" onClick={this.onClick}>
            Delete Expense
          </button>
        </div>
        <DeleteDialogModal 
          clickedDelete={this.state.clickedDelete}
          confirmDelete={this.state.confirmDelete}
          handleClearDeleteDialogModal={this.handleClearDeleteDialogModal}
          description={this.props.expense.description}
        />
      </div>
    );
  }
}

const mapStateToProp = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: data => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProp, mapDispatchToProps)(EditExpensePage);
