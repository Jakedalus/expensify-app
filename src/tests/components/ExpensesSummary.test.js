import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total'; 

import expenses from '../fixtures/expenses';

test('should correctly display 1 expense', () => {
    const wrapper = shallow(
        <ExpensesSummary expenses={[expenses[0]]} getTotal={selectExpensesTotal} />
    );
    expect(wrapper).toMatchSnapshot();
});

test('should correctly display multiple expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary expenses={expenses} getTotal={selectExpensesTotal} />
    );
    expect(wrapper).toMatchSnapshot();
});

