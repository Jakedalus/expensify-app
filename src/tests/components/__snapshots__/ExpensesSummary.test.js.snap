// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly display 1 expense 1`] = `
<div>
  <h1>
    Viewing 
     expenses 
    totalling 
    $0.00
  </h1>
</div>
`;

exports[`should correctly display multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
     expenses 
    totalling 
    $0.00
  </h1>
</div>
`;
